python/ray/serve/tests/test_cli.py

```diff
@@ -222,5 +223,85 @@ def test_info(ray_start_stop):
     )
 
 
+def parrot(request):
+    return request.query_params["sound"]
+
+
+@pytest.mark.skipif(sys.platform == "win32", reason="File path incorrect on Windows.")
+def test_run(ray_start_stop):
+    # Deploys valid config file and import path via serve run
+
+    # Deploy via config file
+    config_file_name = os.path.join(
+        os.path.dirname(__file__), "test_config_files", "two_deployments.yaml"
+    )
+    p = subprocess.Popen(["serve", "run", "-c", config_file_name])
+    time.sleep(4)
+
+    assert requests.get("http://localhost:8000/shallow").text == "Hello shallow world!"
+    assert requests.get("http://localhost:8000/one").text == "2"
+
+    p.send_signal(signal.SIGINT)  # Equivalent to ctrl-C
+    time.sleep(4)
+
+    with pytest.raises(requests.exceptions.ConnectionError):
+        requests.get("http://localhost:8000/shallow")
+    with pytest.raises(requests.exceptions.ConnectionError):
+        requests.get("http://localhost:8000/one")
+
+    # Deploy via import path
+    p = subprocess.Popen(["serve", "run", "-i", "ray.serve.tests.test_cli.parrot"])
+    time.sleep(4)
+
+    assert requests.get("http://localhost:8000/parrot?sound=squawk").text == "squawk"
+
+    p.send_signal(signal.SIGINT)  # Equivalent to ctrl-C
+    time.sleep(4)
```

Comment:
same here; don't use sleep

in general, if you're using `time.sleep` in a test, you should try to rethink the test and see if you can write it in a way that is more robust to timing issues. `time.sleep` will also make tests slower in the common case.

Reply:
Got it, thanks for letting me know. For now, I rewrote the test to use `wait_for_condition`.